rllib/models/torch/torch_action_dist_v2.py

```diff
@@ -0,0 +1,188 @@
+"""The main difference between this and the old action distribution is that this one
+has more explicit input args. So that the input format does not have to be guessed from
+the code. This matches the design pattern of torch distribution which developers may
+already be familiar with.
+"""
+import gym
+import numpy as np
+from typing import Optional
+import abc
+
+
+from ray.rllib.models.action_dist_v2 import ActionDistributionV2
+from ray.rllib.utils.annotations import override, DeveloperAPI
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.typing import TensorType, Union, Tuple, ModelConfigDict
+
+torch, nn = try_import_torch()
+
+
+@DeveloperAPI
+class TorchDistribution(ActionDistributionV2, abc.ABC):
+    """Wrapper class for torch.distributions."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+        self.dist = self._get_distribution(*args, **kwargs)
+
+    @abc.abstractmethod
+    def _get_distribution(self, *args, **kwargs) -> torch.distributions.Distribution:
```

Comment:
Hmm, the more I think about this, we should rename this into `_get_torch_distribution` for clarity (vs `_get_tf_distribution`).
Or even `_get_underlying_torch_distribution`. B/c `self` is already a `TorchDistribution`, albeit an RLlib one :)

Reply:
I agree. fixed.